string script 
string args 
string shell 
---
bool success 
string message
